### PR TITLE
Use consistent variable style in shell examples

### LINKED ISSUE #552: Tidy up DEV_GUIDE shell examples and document additional filters profile
> We've got a bit of inconsistency with placeholders in the text.  We've got curly brackets here and dollars used down below `$your_quay_username$` etc.  We should use a single style.  I don't care which.

_Originally posted by @k-wall in https://github.com/kroxylicious/kroxylicious/pull/542#discussion_r1293108667_

### PR DESCRIPTION
**Pull request by robobario**
Resolves #552

### Type of change

- Documentation

### Description

Use consistent variable style in shell examples

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by robobario** [comment]
Ah just noticed there was a second bit in the issue to document withAdditionalFilters, will do that.

File: DEV_GUIDE.md [hunk 1]
```diff
@@ -60,6 +60,14 @@ Build the project like this:
 mvn clean install
 ```
 
+The project provides some Kroxylicious-maintained Filter implementations that are not built by default. To also build
```
Comment (SamBarker) [inline DEV_GUIDE.md:63]: Possibly worth clarifying they are not included in distribution by default we do compile them.

**Review by SamBarker — APPROVED** [review]
LGTM

File: DEV_GUIDE.md (hunk 1 above)
Comment (robobario) [reply DEV_GUIDE.md]: Cool, I didn't realise this, have moved it down to be under the section describing the `dist` profile.

**Pull request merged**